debug/js/main.js

```diff
@@ -1 +1,65 @@
+(function(muxjs) {
+  'use strict';
 
+  var diffParsed = function(expected, actual) {
+    var expectedLines = muxjs.tools.textifyMp4(expected, null, ' ').split('\n');
+    var actualLines = muxjs.tools.textifyMp4(actual, null, ' ').split('\n');
+    var matcher = new difflib.SequenceMatcher(expectedLines, actualLines);
+
+    return diffview.buildView({
+      baseTextLines: expectedLines,
+      newTextLines: actualLines,
+      opcodes: matcher.get_opcodes(),
+      baseTextName: "Expected MP4",
+      newTextName: "Actual MP4",
+      contextSize: 10,
+      viewType: 0
+    });
+  };
+
+  var logMediaSource = function(event) {
+    console.log('media source', event.type);
+  };
+  var logSourceBuffer = function(event) {
+    console.log('source buffer', event.type);
+  };
+
+  var prepareVideo = function(video, options, callback) {
+    var mediaSource  = new MediaSource();
+
+    if (typeof options === 'function') {
+      callback = options;
+      options = {};
+    }
+
+    video.src = URL.createObjectURL(mediaSource);
+    mediaSource.addEventListener('error', logMediaSource);
+    mediaSource.addEventListener('opened', logMediaSource);
+    mediaSource.addEventListener('closed', logMediaSource);
+    mediaSource.addEventListener('sourceended', logMediaSource);
+
+    mediaSource.addEventListener('sourceopen', function () {
+      var sourceBuffer, codecs;
+
+      codecs = options.codecs || 'avc1.64001f,mp4a.40.2';
+
+      sourceBuffer = mediaSource.addSourceBuffer('video/mp4;codecs="' + codecs + '"');
+
+      sourceBuffer.addEventListener('updatestart', logSourceBuffer);
+      sourceBuffer.addEventListener('updateend', logSourceBuffer);
+      sourceBuffer.addEventListener('error', logSourceBuffer);
+
+      video.addEventListener('error', logSourceBuffer);
+      video.addEventListener('error', function() {
+        video.classList.add('error');
+      });
+
+      return callback(mediaSource, sourceBuffer);
+    });
+  }
+
+  muxjs.debug = {
```

Comment:
I don't like that these helper functions are in the muxjs "namespace".

I'm even second guessing our inclusion of mp4-inspector and flv-inspector in the namespace since they are only used in these debugging pages and are not even part of what would be the mux.js "build".